### ISSUE TITLE
Device status in the API

### ISSUE THREAD

**Issue by imrehg** [resin-io/resin-sdk#183]
I was trying out the SDK, and running the demo `resin.models.device.getAll()` request. All my devices are offline at the moment and the result of the query for the device status are all like this:

```
status: 'Idle',
is_online: false,
```

Using the dashboard directly, these are the status possibilities:

* `Idle`
* `Configuring`
* `Updating`
* `Offline`
* `Post Provisioning`

and all devices are marked as `Status: Offline`. Isn't that what the API should return too, instead of `Idle`? Or the meaning of the words are different for the API and the dashboard? Both are acceptable answers, of course, just checking if this is an API mismatch or a lack of documentation or something else.

**Comment by emirotin** [comment]
The API and thus SDK return raw data, while dashboard does some interpreting. Obviously we would like to normalize it in the future, but for now that's whay we have. The status detection code in the dashboard is roughly this:

```
			seen = device.last_seen_time? and moment(device.last_seen_time).year() >= 2013
			return Status.POST_PROVISIONING if device.provisioning_state is 'Post-Provisioning'
			return Status.CONFIGURING if not device.is_online and not seen
			return Status.OFFLINE if not device.is_online
			return Status.UPDATING if device.download_progress? and device.status is 'Downloading'
			return Status.CONFIGURING if device.provisioning_progress?
			return Status.IDLE
```

To reiterate: parts of this can change in the future, but this code will eventually migrate to the SDK and will be exposed as a public function.

**Comment by imrehg** [comment]
Okay, then it seems a documentation issue, the case that the two "status" variables are not equivalent. 

That logic flow makes sense, thanks for the details! I see it does not use the `status` returned by the API at all, though. Does it mean in the SDK the `device.status` returned should just be ignored for the time being?

**Comment by emirotin** [comment]
I agree this should be documented and normalized across our repos. As we move to open-sourcing more of our code that definitely will happen.

Well, the `status` was poorly named as is actually the "sub-status". You see it in the dashboard in specific cases.
Precisely the text you see is (`status => displayText`):

```
Status.OFFLINE => 'Offline'
Status.UPDATING => 'Updating…'
Status.CONFIGURING => device.provisioning_state || 'Configuring…'
Status.POST_PROVISIONING => 'Post Provisioning…'
otherwise => device.status || 'Idle'
```

@jviotti can you add these two methods (`getStatus` [https://github.com/resin-io/resin-sdk/issues/183#issuecomment-174883640] and `getStatusText` defined here) to the device model module of the SDK? We will later reuse it in the dashboard.

**Comment by jviotti** [comment]
@imrehg @emirotin I've been actually working on extracting this logic to a separate module yesterday, which will then be reused by the SDK. I estimate I'll finish with that today.

I'll keep you updated on this issue!

**Comment by jviotti** [comment]
I've implemented `resin.models.device.getStatus()` here: https://github.com/resin-io/resin-sdk/pull/184

**Comment by jviotti** [comment]
@imrehg I've released v5.1.0 which includes `resin.models.device.getStatus()`.
